hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -89,18 +91,17 @@ long getLastCallId() {
      * Registers an invocation.
      *
      * @param invocation The invocation to register.
+     * @return true if InvocationRegistry is alive when invocation is registered, false otherwise
```

Comment:
another idea: couldn't the `register()` method itself notify the invocation with the `HazelcastInstanceNotActiveException`  - It's the responsibility of the `InvocationRegistry` to notify invocations on shutdown anyway. 

Then you can simply change the contract to "returns false when InvocationRegistry is not alive and registration is not successful. "